src/xrpld/core/detail/DatabaseCon.cpp

```diff
@@ -237,6 +237,40 @@ setup_DatabaseCon(Config const& c, std::optional<beast::Journal> j)
     }
     setup.useGlobalPragma = true;
 
+    auto setPragma =
+        [](std::string& pragma, std::string const& key, int64_t value) {
+            pragma = "PRAGMA " + key + "=" + std::to_string(value) + ";";
+        };
+
+    // Lgr Pragma
+    setPragma(setup.lgrPragma[0], "journal_size_limit", 1582080);
+
+    // TX Pragma
+    int64_t page_size = 4096;
+    int64_t journal_size_limit = 1582080;
+    int64_t max_page_count = 4294967294;
+
+    if (c.exists("sqlite"))
+    {
+        auto& s = c.section("sqlite");
+        set(page_size, "page_size", s);
+        set(journal_size_limit, "journal_size_limit", s);
+    }
```

Comment:
needs to enforce valid `page_size` values, as per https://www.sqlite.org/pragma.html#pragma_page_size